test/helpers/kubectl.go

```diff
@@ -795,6 +797,119 @@ func (kub *Kubectl) GetCiliumPodOnNode(namespace string, node string) (string, e
 	return res.Output().String(), nil
 }
 
+// isPodFinished return true if the pod has finish up running.
+func (kub *Kubectl) isPodFinished(ns, podName string) (bool, error) {
+	pod, err := kub.CoreV1().Pods(ns).Get(podName, metav1.GetOptions{})
+	if err != nil {
+		return false, err
+	}
+	switch pod.Status.Phase {
+	case v1.PodFailed, v1.PodSucceeded:
+		return true, nil
+	}
+	return false, nil
+}
+
+// WaitForPodFinish waits for the given pod in the given namespace until the pod
+// is finish running or until the given context deadline is reached.
+func (kub *Kubectl) WaitForPodFinish(ctx context.Context, ns, podName string) error {
+	for {
+		finish, err := kub.isPodFinished(ns, podName)
+		if err != nil {
+			return err
+		}
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		default:
+			if finish {
+				return nil
+			}
+			time.Sleep(time.Second)
+		}
+	}
+}
+
+// isPodReady returns true if the given pod int he given namespace is running
+// and in ready state.
+func (kub *Kubectl) isPodReady(ns, podName string) (bool, error) {
+	pod, err := kub.CoreV1().Pods(ns).Get(podName, metav1.GetOptions{})
+	if err != nil {
+		return false, err
+	}
+	switch pod.Status.Phase {
+	case v1.PodFailed, v1.PodSucceeded:
+		return false, errors.New("Pod is not running")
+	case v1.PodRunning:
+		for _, cond := range pod.Status.Conditions {
+			if cond.Type == v1.PodReady && cond.Status == v1.ConditionTrue {
+				return true, nil
+			}
+		}
+	}
+	return false, nil
+}
+
+// WaitForPodReady waits for the given pod in the given namespace until the pod
+// is running and in ready state or until the given context deadline is reached.
+func (kub *Kubectl) WaitForPodReady(ctx context.Context, ns, podName string) error {
+	for {
+		ready, err := kub.isPodReady(ns, podName)
+		if err != nil {
+			return err
+		}
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		default:
+			if ready {
+				return nil
+			}
+			time.Sleep(time.Second)
+		}
+	}
+}
+
+// isPodSuccess returns true if the pod has finish up running. Error is returned
+// if an unsuccessfully exit has occurred.
+func (kub *Kubectl) isPodSuccess(ns, podName string) (bool, error) {
```

Comment:
unsuccessfully -->  unsuccessful

Reply:
why no `helpers.WaitForPods()` and added a new function?

Reply:
WaitForPods waits for pods to be in Ready state, this waits for pod finish / exit with a success.

For example, if I have a pod running with `sleep 1000s`, `WaitForPods` will return true as soon the pod is running. `isPodSuccess` returns true after `1000s`